### PR TITLE
Update compileSdkVersion to 25

### PR DESCRIPTION
**Pull request by NightlyNexus**

### CODE REVIEW

File: telescope/src/main/java/com/mattprecious/telescope/EmailDeviceInfoLens.java [hunk 1]
```diff
@@ -15,14 +15,14 @@
...
  * <ul>
- *   <li>App version</li>
- *   <li>App version code</li>
- *   <li>Device manufacturer</li>
- *   <li>Device model</li>
- *   <li>Screen resolution</li>
- *   <li>Screen density</li>
- *   <li>Android version</li>
- *   <li>Android API level</li>
+ * <li>App version</li>
```
Comment (mattprecious) [inline telescope/src/main/java/com/mattprecious/telescope/EmailDeviceInfoLens.java]: Revert

File: telescope/src/main/java/com/mattprecious/telescope/EmailDeviceInfoLens.java [hunk 2]
```diff
@@ -36,9 +36,8 @@
    * @deprecated Use {@link #EmailDeviceInfoLens(Context, String, String...)} or {@link
    * #EmailDeviceInfoLens(Context, String, String, int, String...)}.
    */
-  @Deprecated
-  public EmailDeviceInfoLens(Context context, String[] addresses, String subject, String version,
-      int versionCode) {
+  @Deprecated public EmailDeviceInfoLens(Context context, String[] addresses, String subject,
```
Comment (mattprecious) [inline telescope/src/main/java/com/mattprecious/telescope/EmailDeviceInfoLens.java]: This, too.

**Comment by NightlyNexus** [comment]
(I'm sorry; I can't figure out what's failing. `./gradlew clean check` works for me locally. If anybody knows what's wrong here, I'll fix it.)

**Comment by JakeWharton** [comment]
JDK 7 needs removed from `.travis.yml` since Java 8 is required to build API 25. As for the Java 8 build, there's a failing lint check. Add `lintOptions` which prints failures to standard out and enables the text report (check something like SQL Brite for the config).

File: .travis.yml [hunk 3]
```diff
@@ -1,15 +1,14 @@
 language: android
 
 jdk:
- - oraclejdk7
  - oraclejdk8
 
 android:
   components:
-    - platform-tools
     - tools
```
Comment (NightlyNexus) [inline .travis.yml]: Sorry about the notification spam!
Before this change, the error said there was an old platform-tools version which could not check 25.

File: build.gradle [hunk 4]
```diff
@@ -1,7 +1,7 @@
 allprojects {
   buildscript {
     repositories {
-      mavenCentral()
+      jcenter()
```
Comment (mattprecious) [inline build.gradle]: 😞

Comment (NightlyNexus) [inline build.gradle]: Yeah, nothing we can do, I don't think. :(

**Pull request merged**